kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp

```diff
@@ -151,6 +151,14 @@ pcl::KdTreeFLANN<PointT, Dist>::nearestKSearch (const PointT &point, int k,
                            k_indices_mat, k_distances_mat,
                            k, param_k_);
 
+  const size_t& max_index = identity_mapping_ ? total_nr_points_ : index_mapping_.size();
+  k_indices.erase(std::remove_if(k_indices.begin(), k_indices.end(),
+                                 [max_index](int idx) {
```

Comment:
Replace `int` with `index_t` in this function

Reply:
I used `int` because it's the type of the `k_indices` argument. Should the whole function also be updated to use `index_t`?

Reply:
int and index_t are the same (until we switch on 1.12), but index_t is chosen at compile time, not based on architecture

Reply:
OK, I updated just this lambda to use `index_t`.